config/package-is-installable/src/getSystemNodeVersion.ts

```diff
@@ -3,7 +3,7 @@ import * as execa from 'execa'
 
 export function getSystemNodeVersionNonCached () {
   // @ts-expect-error
-  if (process['pkg'] != null) {
+  if (process['pkg'] == null) {
     return execa.sync('node', ['--version']).stdout.toString()
   }
```

Comment:
You are breaking node.js check on systems that have node. This is the way we need to check the version of node.js in the packages version of pnpm (where we bundle pnpm with node.js)

The right solution is probably to do something like
```suggestion
  if (process['pkg'] != null) {
    try {
      return execa.sync('node', ['--version']).stdout.toString()
    } catch (err) {
      // There is no Node.js on the system
      return null
    }
  }
```